tests/FSharpPlus.Tests/General.fs

```diff
@@ -1544,10 +1544,18 @@ module Applicative =
         let run (ZipList x) = x
         let run' (ZipList' x) = x
 
+
         // Test Applicative (functions)
+        let fn = 
+            let mutable x = 0
+            fun () -> 
+                x <- x + 1
+                x
         let res607 = map (+) ( (*) 100 ) 6 7
         let res606 = ( (+) <*>  (*) 100 ) 6
         let res508 = (map (+) ((+) 3 ) <*> (*) 100) 5
+        let res123 = (map tuple3 fn <*> fn <*> fn) ()
+        let res456 = (sequence [fn; fn; fn] : unit -> int list) ()
```

Comment:
I wasn't sure whether it's appropriate to put this more integration-level test here inside the test module for Applicative, or if it should be in a test module for Traversable, or if we shouldn't bother with a test like this.